src/test/java/edu/cornell/kfs/vnd/batch/service/impl/CuCommodityCodeUpdateServiceImplTest.java

```diff
@@ -1,45 +1,56 @@
-import java.util.ArrayList;
-import org.apache.commons.io.FileUtils;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotSame;
+import static org.junit.Assert.assertTrue;
 
-import org.kuali.kfs.sys.ConfigureContext;
-import org.kuali.kfs.sys.context.KualiTestBase;
-import org.kuali.kfs.sys.context.SpringContext;
+import java.util.ArrayList;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
 import org.kuali.kfs.vnd.businessobject.CommodityCode;
-import org.kuali.rice.core.api.config.property.ConfigurationService;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
 
-import edu.cornell.kfs.vnd.batch.service.CommodityCodeUpdateService;
+import edu.cornell.kfs.sys.batch.service.impl.CuBatchInputFileServiceImpl;
+import edu.cornell.kfs.vnd.batch.CommodityCodeInputFileType;
 
-@ConfigureContext
-public class CuCommodityCodeUpdateServiceImplTest extends KualiTestBase {
+public class CuCommodityCodeUpdateServiceImplTest {
 	
-	private CommodityCodeUpdateService commodityCodeUpdateService;
-    private ConfigurationService  kualiConfigurationService;
+	private CommodityCodeUpdateServiceImpl commodityCodeUpdateService;
 
 	private static final String DATA_FILE_PATH = "src/test/resources/edu/cornell/kfs/vnd/fixture/commodityCodeFlatFile.txt";
+	private static final int NUMBER_OF_RECORDS_IN_DATA_FILE = 53321;
     
-	@Override
-	protected void setUp() throws Exception {
-		super.setUp();
-		commodityCodeUpdateService = SpringContext.getBean(CommodityCodeUpdateService.class);
+	@Before
+	public void setUp() throws Exception {
+		commodityCodeUpdateService = PowerMockito.spy(new CommodityCodeUpdateServiceImpl());
+		PowerMockito.doReturn(true).when(commodityCodeUpdateService, "updateCommodityCodes", Mockito.any());
+		
+		CuBatchInputFileServiceImpl batchInputFileService = new CuBatchInputFileServiceImpl();
+		commodityCodeUpdateService.setBatchInputFileService(batchInputFileService);
+		
+		CommodityCodeInputFileType commodityCodeInputFileType = new CommodityCodeInputFileType();
+		commodityCodeUpdateService.setCommodityCodeInputFileType(commodityCodeInputFileType);
 	}
 	
+	@After
+	public void tearDown() {
+	    commodityCodeUpdateService = null;
+	}
+	
+	@Test
 	public void testLoadCommodityCodeFile() {
-		byte[] zeroLength = {};
-		ArrayList<CommodityCode> noList = new ArrayList();
-		byte[] noContent = commodityCodeUpdateService.getFileContent("no such file");
-		ArrayList<CommodityCode> noCommodityList = (ArrayList<CommodityCode>) commodityCodeUpdateService.parseCommodityCodeList(zeroLength);
 		byte[] content = commodityCodeUpdateService.getFileContent(DATA_FILE_PATH);
 		ArrayList<CommodityCode> someCommodityCodes = (ArrayList<CommodityCode>) commodityCodeUpdateService.parseCommodityCodeList(content);
 		
-		assertEquals(noContent.length, zeroLength.length);
-		assertEquals(noCommodityList.size(), noList.size());
 		assertNotSame(content.length, 0);
```

Comment:
Should this test be using "assertNotSame"? I think that does a "object1 != object2" check instead of a "!object1.equals(object2)" check. (Note that "assertNotSame" only has Object-comparing variants, so primitives might be getting "boxed" prior to comparsion.) It may be better to use "assertNotEquals" instead.

Reply:
thanks for the info, I switched